src/AppCenterExtensions.XamarinForms/PageExtensions.cs

```diff
@@ -25,13 +29,45 @@ public static void TrackPage(this Page page, TimeSpan duration, IAnalytics analy
             var properties = new Dictionary<string, string>
             {
                 [nameof(Page.Title)] = page.Title,
-                ["Duration"] = duration.TotalSeconds.ToString(CultureInfo.InvariantCulture)
+                [DurationKey] = duration.TotalSeconds.ToString(CultureInfo.InvariantCulture)
             };
 
+            foreach (var additionalTrackingInfo in AdditionalTrackingInfo)
+            {
+                if (properties.ContainsKey(additionalTrackingInfo.Key))
+                {
+                    properties[additionalTrackingInfo.Key] = additionalTrackingInfo.Value;
+                }
+                else
+                {
+                    properties.Add(additionalTrackingInfo.Key, additionalTrackingInfo.Value);
+                }
+            }
+
             (analytics ?? AppCenterAnalytics.Instance)
                 .TrackEvent(
                     page.GetType().Name.ToTrackingEventName(),
                     properties);
         }
+
+        /// <summary>
+        /// Track any additonal information.
+        /// </summary>
+        /// <param name="key">Key of the additional information</param>
+        /// <param name="value">Value of the additional information</param>
+        public static void AddAdditionalTrackingInfo(string key, string value)
+        {
+            if (key == DurationKey)
+                return;
+
+            if (AdditionalTrackingInfo.ContainsKey(key))
+            {
+                AdditionalTrackingInfo[key] = value;
```

Comment:
This is just me nit picking...

`AdditionalTrackingInfo[key] = value;` alone should suffice as the dictionary indexer will create the key if it doesn't exist, and will update it if it does. See this demonstrated in[ this .NET Fiddle](https://dotnetfiddle.net/v3nBDF)

The changes here are good enough and I will approve as soon as I can see the build succeeding

Reply:
Interesting, from my past experience that used to throw an error, but I have just tried your approach on my side and it works.